Library/AppEnvironment.swift

```diff
@@ -87,36 +89,36 @@ public struct AppEnvironment {
   }
 
   // Replace the current environment with a new environment.
-  public static func replaceCurrentEnvironment(env: Environment) {
+  public static func replaceCurrentEnvironment(_ env: Environment) {
     pushEnvironment(env)
-    stack.removeAtIndex(stack.count - 2)
+    stack.remove(at: stack.count - 2)
   }
 
   // Pushes a new environment onto the stack that changes only a subset of the current global dependencies.
   public static func pushEnvironment(
-    apiService apiService: ServiceType = AppEnvironment.current.apiService,
-               apiDelayInterval: NSTimeInterval = AppEnvironment.current.apiDelayInterval,
+    apiService: ServiceType = AppEnvironment.current.apiService,
+               apiDelayInterval: DispatchTimeInterval = AppEnvironment.current.apiDelayInterval,
```

Comment:
i've opted to use `DispatchTimeInterval` for our environment times instead of `TimeInterval`. it allows you to specify seconds/milliseconds/microseconds/nanoseconds as integers instead of only seconds as a double.